services/sharedchannel/service.go

```diff
@@ -4,28 +4,27 @@
 package sharedchannel
 
 import (
+	"errors"
 	"fmt"
 	"net/url"
 	"sync"
 	"time"
 
 	"github.com/mattermost/mattermost-server/v5/mlog"
 	"github.com/mattermost/mattermost-server/v5/model"
+	"github.com/mattermost/mattermost-server/v5/services/filesstore"
 	"github.com/mattermost/mattermost-server/v5/services/remotecluster"
 	"github.com/mattermost/mattermost-server/v5/store"
 )
 
 const (
 	TopicSync                    = "sharedchannel_sync"
 	TopicChannelInvite           = "sharedchannel_invite"
-	MaxConcurrentUpdates         = 5
+	TopicUploadCreate            = "sharedchannel_upload"
 	MaxRetries                   = 3
-	MaxPostsPerSync              = 50
+	MaxPostsPerSync              = 12          // a bit more than one typical screenfull of posts
+	AsyncAttachmentSizeThreshold = 1000 * 1024 // 1MB - anything bigger is sent asynchronously
```

Comment:
This is just me being pedantic but I'd stick with traditional IEC units. 1MB being 2^20 bytes (1024 * 1024).

Reply:
I'm selling hardrives. :smile:    Actually, that's a good catch because this is left over from when I thought I could send some larger attachments lazily after the post but that won't work without refactoring app layer.   Removed.

Reply:
Fixed via https://github.com/mattermost/mattermost-server/pull/16880/commits/cc4bba95fe068054cc53451e3b2e533373bb1763